Modernize and Refurbish

pajacobson opened the pull request:
- Excuse the newb Ruby code in `sim_test_fixture.rb`
- Updated to ceedling 0.31.1
- Updated to use MDB

pajacobson committed:
Update project.yml

pajacobson committed:
Update to use mdb

pajacobson committed:
Remove archaic files

pajacobson committed:
Update vendor folder to Ceedling 0.31.1
2012 is soooo last decade.

pajacobson committed:
Update for mdb

pajacobson committed:
Updates to test_fixture

pajacobson committed:
Note XC version

pajacobson committed:
Use environment to set paths, compiler

mvandervoord commented:
Thanks for taking a look at this. I often forget about these little peripheral projects.

To simplify support for this one, we could remove all of Ceedling, etc from the repo itself and update this project.yml file to use the gem version of ceedling.

pajacobson committed:
Updated files

pajacobson committed:
Remove vendor

pajacobson commented:
> Thanks for taking a look at this. I often forget about these little peripheral projects.
> 
> To simplify support for this one, we could remove all of Ceedling, etc from the repo itself and update this project.yml file to use the gem version of ceedling.

I removed the vendor folder and ceedling.
project.yml is new, and I've tried to make minimal changes to the default.

I *finally* figured out how to make the throwtheswitch.org/build/cross code work, so the sim_test_fixture has been updated to incorporate that.  For some reason the PIC24 isn't triggering the breakpoint on the UnityHelperDeadLoop. 

I've changed the "loop" to while(0){};  because the XC16 code hangs up otherwise.
PIC32/XC32 triggers the breakpoint OK with this arrangement so there appears to be correct behaviour as far as I've tested.

sim_test_fixture intercepts STDOUT and STDERR.  
MDB displays errors relating to Java logr and NetBeans preferences every time it's launched.
That is super annoying and pointless so the code is filtering those from STDERR output.

pajacobson committed:
Change output extension to .elf
- Allows import into MPLABX IDE

pajacobson committed:
Move INCLUDE_UNITY_CONFIG_H under test defines

pajacobson committed:
Update comments

pajacobson committed:
Revert to loop behavior

pajacobson committed:
Output cleaned errors to STDERR

pajacobson committed:
Remove testing TEST_FAIL

pajacobson committed:
White space changes

pajacobson committed:
Adding missing debug defines

pajacobson committed:
Move fixture config to project.yml

pajacobson committed:
Remove config file

pajacobson committed:
Formatting

pajacobson committed:
Formatting fixup
